### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by CodyRoy**

### CODE REVIEW

File: app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianInterface.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.cr554.newsapp.GuardianAPI;
+
+/**
+ * Created by cr554 on 2/20/2017.
+ */
+
+public interface GuardianInterface {
+
+}
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianInterface.java:9]: Huh? Why add an empty interface?

File: app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianResponse.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public List<GuardianResult> getResults() {
+        return results;
+    }
+
+    public void setResults(List<GuardianResult> results) {
+        this.results = results;
+    }
+}
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianResponse.java:83]: The naming is confusing. Whats the purpose of this class and the one below?

File: app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianService.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.Volley;
+
+/**
+ * Created by cr554 on 2/20/2017.
+ */
+
+public class GuardianService {
+    public static final String API_BASE_URL = "http://content.guardianapis.com/";
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianService.java:11]: Better to use assets for static strings. It's much easier to manage them in the future.

File: app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianService.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    RequestQueue queue = Volley.newRequestQueue(this);
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/GuardianAPI/GuardianService.java:12]: RequestQueues should not be created on a per-service / per-class-instance basis. They should, usually, be singletons. Please see this for much more detail (and the reasoning behind why that is the case): https://developer.android.com/training/volley/requestqueue.html#singleton

Here you have an example of such a singleton implemented by me for the tutorial :)

https://github.com/YaleSTC/android_testing_training/blob/master/app/src/main/java/com/example/samplegallery/Utilities/VolleyRequestQueue.java

File: app/src/main/java/com/example/cr554/newsapp/MainActivity.java [hunk 5]
```diff
@@ -2,14 +2,40 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        
+        newsArticles = new ArrayList<>();
+        newsList.setLayoutManager(new LinearLayoutManager(this));
+        NewsAdapter newsAdapter = new NewsAdapter(MainActivity.this, newsArticles);
+        newsList.setAdapter();
+        fetchNews();
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/MainActivity.java:32]: This function exists? I can't seem to find the definition.

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -10,8 +10,21 @@
...
+    <!--this text view is used as a default if no news is found-->
     <TextView
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!" />
+        android:text="@string/noNews"
+        android:visibility="gone"
+        android:gravity="center"
+        android:id="@+id/noNewsToShow"/>
+
```
Comment (stswidwinski) [inline app/src/main/res/layout/activity_main.xml:21]: spacing

File: app/src/main/java/com/example/cr554/newsapp/adapter/NewsAdapter.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        } else{
+            //load image
+        }
+    }
+
+    @Override
+    public int getItemCount(){
+        return this.mNewsArticles.size();
+    }
+
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/adapter/NewsAdapter.java:56]: spacing

File: app/src/main/java/com/example/cr554/newsapp/NewsViewHolder.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package com.example.cr554.newsapp;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/NewsViewHolder.java:9]: spacing

File: app/src/main/java/com/example/cr554/newsapp/MainActivity.java [hunk 9]
```diff
@@ -2,14 +2,40 @@
...
+import android.widget.TextView;
+
+import com.android.volley.RequestQueue;
+import com.example.cr554.newsapp.adapter.NewsAdapter;
+
+import org.w3c.dom.Text;
+
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/MainActivity.java:17]: spacing

**Comment by CodyRoy** [comment]
When I started this project I wanted to go in a different direction, using butterknife and gson. Later on in development I adecided to not use butterknife and eventually decided to drop gson as well because I wanted to focus on the Volley side (as it seemed most pertinent to my future work).

As a result the first few pulls were going in a different direction than the end product. Basically everything in that GaurdianAPI package was going to be used as part of the gson/butterknife integration.

**Pull request merged**